app/code/community/PedroTeixeira/Correios/Model/Carrier/CorreiosMethod.php

```diff
@@ -665,4 +670,202 @@ public function isZipCodeRequired($countryId = null)
     {
         return true;
     }
+
+    /**
+     * Retrieve an average size.
+     * For optimization purposes all tree box sizes are converted in one medium dimension.
+     * Result cant exceed the minimum transportation limits.
+     *
+     * @return PedroTeixeira_Correios_Model_Carrier_CorreiosMethod
+     */
+    protected function _loadMidSize()
+    {
+        $volumeFactor = $this->getConfigData('coeficiente_volume');
+        $volumeTotal = $this->_volumeWeight * $volumeFactor;
+        $pow = round(pow((int) $volumeTotal, (1/3)));
+        $x1 = $this->getConfigData('altura_padrao');
+        $x2 = $this->getConfigData('largura_padrao');
+        $x3 = $this->getConfigData('comprimento_padrao');
+        $this->_midSize = max($pow, $x1, $x2, $x3);
+        return $this;
+    }
+
+    /**
+     * Validate post methods removing invalid services from quotation.
+     * 
+     * @return boolean|PedroTeixeira_Correios_Model_Carrier_CorreiosMethod
+     */
+    protected function _removeInvalidServices()
+    {
+        $this->_loadMidSize();
+        $tmpMethods = $this->_postMethodsExplode;
+        foreach ($tmpMethods as $key => $method) {
+            $isOverSize = ($this->_midSize > $this->getConfigData("validate/serv_{$method}/max/size"));
+            $isOverSize |= ($this->_midSize * 3 > $this->getConfigData("validate/serv_{$method}/max/sum"));
+            $isOverWeight = ($this->_packageWeight > $this->getConfigData("validate/serv_{$method}/max/weight"));
```

Comment:
E você só está checando a soma, a maior medida e peso, mas o que acontece se for uma medida inválida? Os Correios vão retornar erro.